src/services/pocket-relayer.ts

```diff
@@ -623,6 +635,31 @@ export class PocketRelayer {
     }
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async enforceLimits(parsedRawData: Record<string, any>, blockchain: string): Promise<string | LimitError> {
+    if (parsedRawData.method === 'eth_getLogs') {
+      const altruistUrl = String(this.altruists[blockchain])
+      let [{ fromBlock, toBlock }] = parsedRawData.params as [{ fromBlock: string; toBlock: string }]
+
+      // If fromBlock or toBlock is equal to undefined it defaults to latest.
+      if ((fromBlock === 'latest' || fromBlock === undefined) && altruistUrl !== 'undefined') {
+        fromBlock = String(await getBlockNumber(altruistUrl))
+      } else if ((toBlock === 'latest' || toBlock === undefined) && altruistUrl !== 'undefined') {
+        toBlock = String(await getBlockNumber(altruistUrl))
+        // We cannot move forward if there is no altruist available.
+      } else if (
+        (toBlock === 'latest' || toBlock === undefined || fromBlock === 'latest' || fromBlock === undefined) &&
```

Comment:
this is hard to read, you could separate these into variables like `toBlockValid` checking them separately so at the end end on the else if you only do a simpler check like `!toBlockValid && !fromBlockValid`